### PR TITLE
US-#444 changes

### PR DESCRIPTION
**Pull request by snehi23**

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Could you pull in the latest changes from master and resolve the merge conflicts?

how-to: https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/

File: controller/handler/trial.js [hunk 1]
```diff
@@ -68,6 +69,23 @@ function trialView (request, reply) {
...
+            database.sequelize.query(
+             `
+             SELECT jcns.rule
+             FROM trial AS tr
+    	     JOIN stage AS st
+             ON tr.id = st.trialId
+             JOIN join_current_and_next_stages AS jcns
+             ON st.id = jcns.stageId
+             WHERE tr.id = ?
+                `,
```
Comment (ChristianMurphy) [inline controller/handler/trial.js]: lines 74-82 need to be indented one level in compared to line 73 (4 spaces)

File: controller/handler/trial.js [hunk 2]
```diff
@@ -88,7 +107,8 @@ function trialView (request, reply) {
                         'Semicompliant',
                         'Noncompliant'
                     ]
-                })
+                }),
+                endDate: processRule(rules)
```
Comment (ChristianMurphy) [inline controller/handler/trial.js]: use `Array.reduce` to combine the rules together into a single value

ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

File: controller/helper/process-rule.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+/**
+ * @module controller/helper/process-rule
+ */
+
+const moment = require('moment');
+
+/**
+ * A rule processing for calculating end date of patient in trial.
+ * @param {Object} ruleValue - aggregated rule values
```
Comment (ChristianMurphy) [inline controller/helper/process-rule.js]: `ruleValue` is type `{number}` which represents the number of days

File: controller/helper/process-rule.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+/**
+ * @module controller/helper/process-rule
+ */
+
+const moment = require('moment');
+
+/**
+ * A rule processing for calculating end date of patient in trial.
+ * @param {Object} ruleTotal - aggregated rule values
```
Comment (ChristianMurphy) [inline controller/helper/process-rule.js]: `ruleValue` is type `{number}` which represents the number of days

File: controller/handler/trial.js [hunk 5]
```diff
@@ -68,6 +69,23 @@ function trialView (request, reply) {
...
+            ),
+            database.sequelize.query(
+              `
+              SELECT jcns.rule
+              FROM trial AS tr
+    	      JOIN stage AS st
+              ON tr.id = st.trialId
+              JOIN join_current_and_next_stages AS jcns
+              ON st.id = jcns.stageId
+              WHERE tr.id = ?
```
Comment (ChristianMurphy) [inline controller/handler/trial.js]: lines 74-82 need to be indented one level in compared to line 73 (4 spaces)

File: controller/helper/process-rule.js (hunk 4 above)
Comment (snehi23) [inline controller/helper/process-rule.js]: Ok. I will do it. Forgot to update it  after changing implementation of function.

File: controller/handler/trial.js (hunk 5 above)
Comment (snehi23) [inline controller/handler/trial.js]: I made changes as per last comment but its keep on changing. I am sure why.

**Comment by ChristianMurphy** [comment]
resolve merge conflicts and this should be good to go! :smile: 

how-to: https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/

**Comment by ChristianMurphy** [comment]
LGTM

**Comment by snehi23** [comment]
Merging US-#444 to master.

**Comment by snehi23** [comment]
Branch US-#444 merged with master.

**Pull request merged**